include/Engine/Core/TextureLibrary.hpp

```diff
@@ -18,14 +18,20 @@
 #ifndef     __ENGINE_TEXTURE_LIBRARY_HPP__
 # define    __ENGINE_TEXTURE_LIBRARY_HPP__
 
-#include    "Engine/Configuration.hpp"
+#include    <GL/gl.h>
+#include    <string>
+#include    <map>
 #include    "Engine/Core/Library.hpp"
 #include    "Engine/Core/Texture.hpp"
+#include    "Engine/Core/AssimpScene.hpp"
 
 namespace   Engine
 {
-    class ENGINE_EXPORTS TextureLibrary : public Library<Texture *>
+    class       TextureLibrary : public Library<Texture *>
     {
+	private:
+	    std::map<std::string, Texture*> _textureMap;
```

Comment:
must use the m_resources from Library<Texture *>, but not matching so we need to change the Library class to add a key type value in template. Using unsigned int for Material and std::string for Texture / Mesh / Animation and using the shader program enum for the shaderprogram library.